libr/bin/format/elf/elf.c

```diff
@@ -2650,57 +2655,56 @@ static size_t get_num_relocs_approx(ELFOBJ *bin) {
 	return get_num_relocs_dynamic (bin) + get_num_relocs_sections (bin);
 }
 
-static size_t populate_relocs_record_from_dynamic(ELFOBJ *bin, RBinElfReloc *relocs, size_t pos) {
+static size_t populate_relocs_record_from_dynamic(ELFOBJ *bin, RBinElfReloc *relocs, size_t pos, size_t num_relocs) {
 	size_t offset;
-	size_t i = 0;
 	size_t size = get_size_rel_mode (bin->dyn_info.dt_pltrel);
 
-	for (offset = 0; offset < bin->dyn_info.dt_pltrelsz; offset += size) {
-		read_reloc (bin, relocs + pos, bin->dyn_info.dt_pltrel,
-				bin->dyn_info.dt_jmprel + offset - bin->baddr);
-		fix_rva_and_offset_exec_file (bin, relocs + pos);
-		pos++;
-		++i;
+	for (offset = 0; offset < bin->dyn_info.dt_pltrelsz && pos < num_relocs; offset += size) {
```

Comment:
Why did you had pos < num_relocs.
num_relocs is compute with the information inside the dynamic section.
So if `offset < bin->dyn_info.dt_pltrelsz` is false => `pos < num_relocs` is false

Reply:
That's true in theory. As explained above, I added it for 2 reasons:
1) as a defensive measure to avoid overflowing the array in case of something else wrong (weak justification, but it is what happened with the wrong computation)
2) because in case you analyze a particular binary, the relocation section may be soo big that you can't allocate the required memory for all the possible relocations, so we may just read the dynamic relocation entries (see https://github.com/radareorg/radare2/pull/17004/files#diff-7c0443a3dd0679f5b11dfd70dd990d93R2745)